Client/WebFilters/AdblockRustEngine.swift

```diff
@@ -7,6 +7,29 @@ import Foundation
 import BraveCore
 
 extension AdblockEngine {
+  public enum ResourceType: String {
+    case script, xhr
+  }
+  
+  /// Check the rust engine if the request should be blocked given the `sourceURL` and `resourceType`.
+  ///
+  /// - Warning: You must provide a absolute URL (i.e. containing a host) fo r `requestURL` and `sourceURL`
+  public func shouldBlock(requestURL: URL, sourceURL: URL, resourceType: ResourceType) -> Bool {
+    let isThirdParty = requestURL.baseDomain != sourceURL.baseDomain
```

Comment:
if `baseDomain` is eTLD+1, please leave a comment saying so, bc otherwise this will look like an origin check to readers and seem very wrong :)